### DISCUSSION TITLE
JEtsy - a Java implementation of the API

### DISCUSSION THREAD

**Discussion by cmunden** [etsy/open-api#1024] [Show and tell]
I have started writing a Java implementation of version 3 of the API.  Still a work in progress.

**Comment by Gareth064** [comment]
I am doing the with dotnet. It's called EtsySharpApi (EtsySharp was taken by another old repo that isn't being updated)

**Reply by cmunden** [reply]
Hello.  Good to see another developer with an interest in the Etsy API.  I just started the migration effort to move JEtsy from version 2 to the new version 3 with OAuth2.

Is your implementation a complete one?

**Reply by Gareth064** [reply]
Hey. No not complete yet. Is still a private repo for now. I use it one of my apps. I add to it whenever I have a new use case.

Once I get to a more stable place with my own app, I will then spend more time getting the API Library up to date with other endpoints that I don't need yet.

**Reply by cmunden** [reply]
I am doing the exact same thing with JEtsy.  I use it in one of my apps and most of what is implemented so far are the API calls that I need to support my private application.  Good luck with it.  Feel free to reach out if you have questions ever about the semantics of the API.  It could be good to compare notes and help with troubleshooting.

**Reply by Sean190828** [reply]
Hello, Good to see another developer with an interest in the Etsy API. I have completed the upgrade of Etsy from version 2 to the new version 3.
